spec/lucky/route_not_found_handler_spec.cr

```diff
@@ -13,4 +19,13 @@ describe Lucky::RouteNotFoundHandler do
       error_handler.call(context)
     end
   end
+
+  it "takes an optional fallback action" do
+    context = build_context(path: "/non-existent")
+    context.request.method = "GET"
+
+    handler = Lucky::RouteNotFoundHandler.new(fallback: SampleFallbackAction::Index)
+    handler.next = ->(_ctx : HTTP::Server::Context) {}
+    handler.call(context)
```

Comment:
this spec is useless at the moment... Should it even do anything?

Reply:
This took awhile to figure out but it needs a `context.server.response.close` to close the io. Here's some code:

```crystal
  it "takes an optional fallback action" do
    output = IO::Memory.new
    context = build_context_with_io(output, path: "/non-existent")
    context.request.method = "GET"
    handler = Lucky::RouteNotFoundHandler.new
    handler.next = ->(_ctx : HTTP::Server::Context) {}

    handler.call(context)

    context.response.close
    output.to_s.should contain "Last chance"
  end
```

I also think another spec where the method is "POST" and raises instead

Reply:
lol. I actually had that same code, but I missed that `context.response.close` and kept getting an error of unexpected http end. Good call. I'll add those two.